packages/syntax-core/src/Badge/Badge.tsx

```diff
@@ -0,0 +1,44 @@
+import classNames from "classnames";
+import badgeBackgroundColor from "../colors/badgeBackgroundColor";
+import badgeForegroundColor from "../colors/badgeForegroundColor";
+
+import Typography from "../Typography/Typography";
+
+import { type BadgeColor } from "../constants";
+
+/**
+ * Badge is a component to display short text and give additional context to features and other components.
+ */
+const Badge = ({
+  text,
+  color = "primary",
+}: {
+  /**
+   * The text to display inside the badge
+   */
+  text: string;
+  /**
+   * The color of the badge
+   *
+   * @defaultValue "primary"
+   */
+  color?: (typeof BadgeColor)[number];
+}): JSX.Element => (
+  <span
```

Comment:
@jliotta Could we make this a `div` instead which will make it `display: block`? If folks do want to show it next to something else, it'll encourage them to use `flex`